### PR TITLE
feat(styled-library): remove props from html element attributes using shouldForwardProp

### PR DESCRIPTION
**Pull request by abereghici**
Added `shouldForwardProp` function to `styled-library` that's used to remove the props from html element attributes.

Before:
Screenshot 2021-11-22 at 13 42 15

After:
Screenshot 2021-11-22 at 13 42 24

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Comment by SiTaggart** [comment]
Hi @abereghici we've got some component render performance tracking working coming up that I would like in place before we look at this. I would like to measure the impact on render performance, just in case it's expensive.  

Agree though, it's pretty icky looking in the DOM and it'd be nice to tidy up. It won't be long.

File: package.json [hunk 1]
```diff
@@ -116,6 +116,7 @@
     "@storybook/addons": "^6.2.9",
     "@storybook/core": "^6.2.9",
     "@storybook/react": "^6.2.9",
+    "@styled-system/should-forward-prop": "^5.1.5",
```
Comment (TheSisb) [inline package.json:119]: Can we move this dependency to the package.json file in the `packages/paste-libraries/styling` folder please? 

We also pin library dependencies to make sure no odd behavior happens downstream, so feel free to remove the `^`.

File: yarn.lock [hunk 2]
```diff
@@ -46172,7 +46191,7 @@ resolve@^1.17.0:
   languageName: node
   linkType: hard
 
-"styled-system@npm:5.1.5":
+"styled-system@npm:5.1.5, styled-system@npm:^5.1.5":
```
Comment (TheSisb) [inline yarn.lock:46194]: Once we pin and rerun `yarn install` this should sort itself out.

**Review by TheSisb — COMMENTED** [review]
This is a welcome change, thanks!  As Simon said, bear with us a little while as we get our perf metrics sorted out.

File: package.json (hunk 1 above)
Comment (abereghici) [reply package.json]: Hey @TheSisb , I moved this dependency in `styling` package and I fixed the version.

**Comment by SiTaggart** [comment]
Closing in favour of https://github.com/twilio-labs/paste/pull/2285

Thank you for your contribution. Sorry I couldn't pull across your commit. Tests were timing out and I needed to add custom style props to the equation

**Pull request closed without merging**